### PR TITLE
Support for leanote 2.6.1 https://github.com/leanote/leanote

### PR DESCRIPTION
**Pull request by PascalNoisette**
Hi,

Here is a pull request to add support for [Leanote](https://github.com/leanote/leanote). It is a GPLv2 free and open source project to take notes. I switch to leanote as a replacement of evernote and I am very happy with it but there is no clipper for firefox and the one for chrome is kind of old, so it would be great to have support in web-clipper.

The pull request I created is a document service + a builtin image hosting service using the leanote http api with axios.

I tested my pull request in a self hosted dockerized instance of leanote v2.6.1 but it should also work on free or subscription based plan at the hosted https://leanote.com/

Regards,
Pascal

### CODE REVIEW

**Comment by DiamondYuan** [comment]
Sorry, I've only been looking at issus, and haven't been paying attention to pull requests.


For better test case writing and support for more platforms (e.g. safari), please use RequestHelper and IBasicRequestService to send requests.


https://github.com/webclipper/web-clipper/blob/master/src/common/backend/clients/github/client.ts

```
import { IBasicRequestService, IExtendRequestHelper } from '@/service/common/request';
import { RequestHelper } from '@/service/request/common/request';
import { Container } from 'typedi';

const request: IExtendRequestHelper = new RequestHelper({
  baseURL: 'https://api.github.com/',
  headers: {
    Accept: 'application/vnd.github.v3+json',
    Authorization: `token token`,
  },
  request: Container.get(IBasicRequestService),
});
```

File: src/common/backend/imageHosting/leanote/service.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    return 'leanote';
+  };
+
+  uploadImage = async ({ data }: UploadImageRequest) => {
+    const blob = Base64ImageToBlob(data);
+    return this.uploadBlob(blob);
+  };
+
+  uploadImageUrl = async (url: string) => {
+    const res = await axios.get(url, { responseType: 'blob' });
```
Comment (DiamondYuan) [inline src/common/backend/imageHosting/leanote/service.ts:22]: Here you can continue to use axios. I will deal with in the future

**Comment by DiamondYuan** [comment]
For better maintenance, I want to improve the test coverage of the project. Please write test cases like this.


https://github.com/webclipper/web-clipper/blob/master/src/common/backend/clients/github/client.test.ts

**Comment by PascalNoisette** [comment]
Hi @DiamondYuan ,

I made the refactor for testabiliy and wrote a test case.

Regards,
Pascal

**Pull request merged**